File: apps/cli/src/commands/status.ts
import { Command } from "@oclif/command"
import { printUser } from "../print/user"
import { authorize } from "../auth/login"
import { getWorkspace, getApp, getHostUrl } from "../config/config"
import { printWorkspace } from "../print/workspace"
import { printHost } from "../print/host"

export default class Status extends Command {
	static description = "get user and workspace status"

	static flags = {}

	static args = []

	async run(): Promise<void> {
		const hostUrl = await getHostUrl()
		printHost(hostUrl)
		const app = await getApp()
		const workspace = await getWorkspace()
		const user = await authorize()
		printUser(user)
		printWorkspace(app, workspace)
	}
}

File: apps/cli/pkg/command/status.go
package command

import (
	"github.com/thecloudmasters/cli/pkg/auth"
	"github.com/thecloudmasters/cli/pkg/config/host"
	"github.com/thecloudmasters/cli/pkg/config/ws"
	"github.com/thecloudmasters/cli/pkg/print"
)

func Status() error {

	user, err := auth.Check()
	if err != nil {
		return err
	}

	host, err := host.GetHost()
	if err != nil {
		return err
	}

	workspace, err := ws.GetWorkspace()
	if err != nil {
		return err
	}

	print.PrintHost(host)
	print.PrintUser(user)
	print.PrintWorkspace(workspace)

	return nil
}
